src/esmock.d.ts

```diff
@@ -1,2 +1,76 @@
-declare function esmock(path: string, localmock?: any, globalmock?: any): any;
+/**
+ * Mocks imports for the module specified by {@link modulePath}.
+ *
+ * The provided mocks replace the imported modules _fully_.
+ *
+ * @param modulePath The module whose imports will be mocked.
+ * @param parent A URL to resolve specifiers relative to; typically `import.meta.url`.
+ *               If not specified, it will be inferred via the stack, which may not work
+ *               if source maps are in use.
+ * @param mockDefs A mapping of import specifiers to mocked module objects; these mocks will
+ *                 only be used for imports resolved in the module specified by {@link modulePath}.
+ * @param globalDefs A mapping of import specifiers to mocked module objects; these mocks will
+ *                   apply to imports within the module specified by {@link modulePath}, as well
+ *                   as any transitively imported modules.
+ * @param opt
+ * @returns The result of importing {@link modulePath}, similar to `import(modulePath)`.
+ */
+declare function esmock(modulePath: string, parent: string, mockDefs?: Record<string, any>, globalDefs?: Record<string, any>, opt?: esmock.Options): any;
+declare function esmock(modulePath: string, mockDefs?: Record<string, any>, globalDefs?: Record<string, any>, opt?: esmock.Options): any;
+
+declare namespace esmock {
+  interface Options {
+    partial?: boolean | undefined;
+    purge?: boolean | undefined;
```

Comment:
These could be documented; not sure quite what to write as personally I would suggest that people use the named functions to set these instead (i.e. treat options as an internal implementation detail).